fe/fe-core/src/main/java/org/apache/doris/datasource/paimon/source/PaimonScanNode.java

```diff
@@ -379,7 +380,29 @@ public List<org.apache.paimon.table.source.Split> getPaimonSplitFromAPI() {
                     .collect(Collectors.toList())
                     .indexOf(slot.getColumn().getName()))
                     .toArray();
-        ReadBuilder readBuilder = source.getPaimonTable().newReadBuilder();
+        Table paimonTable = source.getPaimonTable();
+        if (scanParams != null && scanParams.incrementalRead()) {
+            Map<String, String> paimonScanParams = new HashMap<>();
+            paimonScanParams.put("scan.snapshot-id", null);
+            paimonScanParams.put("scan.mode", null);
+
+            String startSnapshotId = scanParams.getMapParams().get("startSnapshotId");
+            String endSnapshotId = scanParams.getMapParams().get("endSnapshotId");
+            String incrementalBetweenScanMode = scanParams.getMapParams()
+                    .getOrDefault("incrementalBetweenScanMode", "AUTO");
+
+            String startTimestamp = scanParams.getMapParams().get("startTimestamp");
```

Comment:
I didn't see `startTimestamp`, `endTimestamp` and `incrementalBetweenScanMode` test cases

Reply:
Add please also add some negative cases like invalid snapshot id or timestamp.
And expect certain error message